presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -128,17 +128,22 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
         Set<InternalNode> blockedExactNodes = new HashSet<>();
         boolean splitWaitingForAnyNode = false;
 
-        // todo identify if sorting will cause bottleneck
-        List<HostAddress> sortedCandidates = sortedNodes(nodeMap);
+        List<HostAddress> sortedCandidates = null;
```

Comment:
I'm not sure if I understand but how does this avoid unnecessary sorting. Given this is a local variable, it seems like it would be `null` in the following if-branch so it would still do sorting every time `computeAssignments` is called. What am I missing here?

Reply:
That would be correct in the sense that this avoid no unnecessary work when using `HARD_AFFINITY` or `SOFT_AFFINITY`. However, `NO_PREFERENCE` scheduling doesn't need a sorted list and with this change will not need to perform that sort at all.

Reply:
Makes sense :)

Reply:
Do we really need the `if` branch? It feels like it is always true.

Reply:
It's always true for the first split in the batch being scheduled (when using `HARD_AFFINITY` or `SOFT_AFFINITY` but will be false for all of the rest of the splits being scheduled in the batch (still only sorting the list once).

Reply:
Oh right I'm stupid. It now lives in the for loop.